elastic/kibana
# [Fleet] Improve preconfiguration robustness

## Description 

Resolve #107160 

Currently when we have an error while adding a package to a policy (network error could always happen) we get Fleet in a state without policies without any input and there is no way to recover from that without manually deleting some saved object.

To avoid this, I made a few change to the preconfiguration service:
* I changed the error handling so error happening while adding a package policy are no fatal error
* If an error happens we remove the agent policy saved object so the user can retry the setup again

## How to test

* From a fresh ES and Kibana
* set in your kibana.yml  a wrong registry url `xpack.fleet.registryUrl: 'http://localhost:9999'`
* visit fleet you should see an error in the setup 
* set back your registry url to a correct value
* visit fleet the setup should work and you should get the correct agent policies

@jen-huang @ruflin should we backport this 7.14? it's not a trivial change but I guess it could avoid us a lot of support
